tests/test_lane.py

```diff
@@ -2358,186 +2567,22 @@ def results(wl=works_on_gutenberg_lists, must_be_featured=False):
         both_lists_qu, list_2_clauses = gutenberg_list_2_wl.customlist_filter_clauses(
             list_1_qu,
         )
+        # The query has been modified again -- we've added a second join
+        # against CustomListEntry.
+        assert both_lists_qu != list_1_qu
         both_lists_clauses = list_1_clauses + list_2_clauses
 
         # The combined query matches the work that shows up on
         # both lists.
-        eq_([work.id], _run(both_lists_qu, both_lists_clauses))
+        eq_([work], _run(both_lists_qu, both_lists_clauses))
 
         # If we remove `work` from either list, the combined query
-        # matches nothing. This works even though the materialized
-        # view has not been refreshed.
+        # matches nothing.
         for l in [gutenberg_list, gutenberg_list_2]:
             l.remove_entry(work)
             eq_([], _run(both_lists_qu, both_lists_clauses))
             l.add_entry(work)
 
-    def test_random_sample(self):
```

Comment:
Random sampling was only used to generate grouped feeds, which now happens through the search index.